lldb/source/Plugins/ABI/LoongArch/ABISysV_loongarch.cpp

```diff
@@ -0,0 +1,664 @@
+        float_count == 1 && !is_complex) {
+      return_valobj_sp =
+          GetValObjFromFPRegs(thread, reg_ctx, machine, type_flags, byte_size);
+      return return_valobj_sp;
+    }
+  }
+  return return_valobj_sp;
+}
+
+ValueObjectSP ABISysV_loongarch::GetReturnValueObjectImpl(
+    Thread &thread, CompilerType &return_compiler_type) const {
+  ValueObjectSP return_valobj_sp;
+
+  if (!return_compiler_type)
+    return return_valobj_sp;
+
+  ExecutionContext exe_ctx(thread.shared_from_this());
+  return GetReturnValueObjectSimple(thread, return_compiler_type);
+}
+
+bool ABISysV_loongarch::CreateFunctionEntryUnwindPlan(UnwindPlan &unwind_plan) {
+  unwind_plan.Clear();
+  unwind_plan.SetRegisterKind(eRegisterKindDWARF);
+
+  uint32_t pc_reg_num = loongarch_dwarf::dwarf_gpr_pc;
+  uint32_t sp_reg_num = loongarch_dwarf::dwarf_gpr_sp;
+  uint32_t ra_reg_num = loongarch_dwarf::dwarf_gpr_ra;
+
+  UnwindPlan::RowSP row(new UnwindPlan::Row);
+
+  // Define CFA as the stack pointer
+  row->GetCFAValue().SetIsRegisterPlusOffset(sp_reg_num, 0);
+
+  // Previous frame's pc is in ra
+
+  row->SetRegisterLocationToRegister(pc_reg_num, ra_reg_num, true);
+  unwind_plan.AppendRow(row);
+  unwind_plan.SetSourceName("loongarch function-entry unwind plan");
+  unwind_plan.SetSourcedFromCompiler(eLazyBoolNo);
+
+  return true;
+}
+
+bool ABISysV_loongarch::CreateDefaultUnwindPlan(UnwindPlan &unwind_plan) {
+  unwind_plan.Clear();
+  unwind_plan.SetRegisterKind(eRegisterKindGeneric);
+
+  uint32_t pc_reg_num = LLDB_REGNUM_GENERIC_PC;
+  uint32_t fp_reg_num = LLDB_REGNUM_GENERIC_FP;
+
+  UnwindPlan::RowSP row(new UnwindPlan::Row);
+
+  // Define the CFA as the current frame pointer value.
+  row->GetCFAValue().SetIsRegisterPlusOffset(fp_reg_num, 0);
+  row->SetOffset(0);
+
+  int reg_size = 4;
+  if (m_is_la64)
+    reg_size = 8;
+
+  // Assume the ra reg (return pc) and caller's frame pointer
+  // have been spilled to stack already.
+  row->SetRegisterLocationToAtCFAPlusOffset(fp_reg_num, reg_size * -2, true);
+  row->SetRegisterLocationToAtCFAPlusOffset(pc_reg_num, reg_size * -1, true);
+
+  unwind_plan.AppendRow(row);
+  unwind_plan.SetSourceName("loongarch default unwind plan");
+  unwind_plan.SetSourcedFromCompiler(eLazyBoolNo);
+  unwind_plan.SetUnwindPlanValidAtAllInstructions(eLazyBoolNo);
+  return true;
+}
+
+bool ABISysV_loongarch::RegisterIsVolatile(const RegisterInfo *reg_info) {
+  return !RegisterIsCalleeSaved(reg_info);
+}
+
+bool ABISysV_loongarch::RegisterIsCalleeSaved(const RegisterInfo *reg_info) {
+  if (!reg_info)
+    return false;
+
+  const char *name = reg_info->name;
+  ArchSpec arch = GetProcessSP()->GetTarget().GetArchitecture();
+  uint32_t arch_flags = arch.GetFlags();
+  // floating point registers are only callee saved when using
+  // F or D hardware floating point ABIs
+  bool is_hw_fp = (arch_flags & ArchSpec::eLoongArch_abi_mask) != 0;
+
+  bool is_callee_saved = llvm::StringSwitch<bool>(name)
```

Comment:
Just `return llvm::StringSwitch...` instead of making a variable for it.

Reply:
it's done.